modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/MuleSdkOperationModelParserSdk.java

```diff
@@ -226,8 +227,14 @@ public Optional<ExceptionHandlerModelProperty> getExceptionHandlerModelProperty(
 
   @Override
   public Optional<DeprecationModel> getDeprecationModel() {
-    // TODO: MULE-20083
-    return empty();
+    return getSingleChild(operation, "deprecated").map(this::buildDeprecationModel);
+  }
+
+  private DeprecationModel buildDeprecationModel(ComponentAst deprecationAst) {
```

Comment:
move this to an utils class since sources and parameters will need the same thing